x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,150 @@
+		return err
+	}
+
+	// Since the misbehaviour packet was received within the trusting period
+	// w.r.t to the last trusted consensus it entails that the infraction age
+	// isn't too old. see ibc-go/modules/light-clients/07-tendermint/types/misbehaviour_handle.go
+
+	// construct a ligth client attack evidence
+	evidence, err := k.ConstructLightClientEvidence(ctx, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	// jail and tombstone the byzantine validators
+	for _, v := range evidence.ByzantineValidators {
+		// convert consumer consensus address
+		consuAddr := sdk.ConsAddress(v.Address.Bytes())
+		provAddr := k.GetProviderAddrFromConsumerAddr(ctx, misbehaviour.Header1.Header.ChainID, types.NewConsumerConsAddress(consuAddr))
+		k.stakingKeeper.ValidatorByConsAddr(ctx, consuAddr)
+		val, ok := k.stakingKeeper.GetValidatorByConsAddr(ctx, provAddr.Address)
+
+		if !ok || val.IsUnbonded() {
+			logger.Error("validator not found or is unbonded", provAddr.String())
+			continue
+		}
+
+		// jail validator if not already
+		if !val.IsJailed() {
+			k.stakingKeeper.Jail(ctx, provAddr.Address)
+		}
+
+		// tombstone validator if not already
+		if !k.slashingKeeper.IsTombstoned(ctx, provAddr.Address) {
+			k.slashingKeeper.Tombstone(ctx, provAddr.Address)
+		}
+
+		// update jail time to end after double sign jail duration
+		k.slashingKeeper.JailUntil(ctx, provAddr.Address, evidencetypes.DoubleSignJailEndTime)
+	}
+
+	logger.Info(
+		"confirmed equivocation",
+		"byzantine validators", evidence.ByzantineValidators,
+	)
+
+	return nil
+}
+
+// ConstructLightClientEvidence constructs and returns a CometBFT Ligth Client Attack(LCA) evidence struct
+// from the given misbehaviour
+func (k Keeper) ConstructLightClientEvidence(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) (*tmtypes.LightClientAttackEvidence, error) {
+
+	// construct the trusted and conflicetd ligth blocks
+	trusted, err := headerToLightBlock(*misbehaviour.Header1)
+	if err != nil {
+		return nil, err
+	}
+	conflicted, err := headerToLightBlock(*misbehaviour.Header2)
+	if err != nil {
+		return nil, err
+	}
+
+	// get common header using the IBC misbehaviour
+	commonHeight, commonTs, commonValset, err := k.GetTrustedInfoFromMisbehaviour(ctx, misbehaviour)
+	if err != nil {
+		return nil, err
+	}
+
+	// construct the LCA evidence by copying the CometBFT constructor
+	// see newLightClientAttackEvidence() in tendermint/light/detector.go
+	ev := tmtypes.LightClientAttackEvidence{
+		ConflictingBlock: conflicted,
+	}
+
+	if ev.ConflictingHeaderIsInvalid(trusted.Header) {
+		ev.CommonHeight = commonHeight
+		ev.Timestamp = commonTs
+		ev.TotalVotingPower = commonValset.TotalVotingPower()
+	} else {
+		ev.CommonHeight = trusted.Header.Height
+		ev.Timestamp = trusted.Header.Time
+		ev.TotalVotingPower = trusted.ValidatorSet.TotalVotingPower()
+	}
+
+	ev.ByzantineValidators = ev.GetByzantineValidators(commonValset, trusted.SignedHeader)
```

Comment:
discard that, i got confused thinking the byzantine vals are picked from trusted.

Reply:
Following up on the way we get the byzantine validators. I think the way the code is right now may punish more than required. 
The Comet view of the evidence is "opinionated" as it trusts its local chain and assumes all vals that signed the "conflicting block" misbehaved and should be punished. But an external viewer, like the ICS provider module, cannot tell which is the correct block.

Let's say we have a LC detecting:
- two blocks at same height `H` 
  - `B1` signed by `{v2, v3}`, block retrieved from some `node1`
  - `B2` signed by `{v3, v4}`, block retrieved from another `node2`
 - the common header at height `T` signed by `{v1, v2, v3, v4}`

The LC agent will submit two evidences of form  `ev(conflictingBlock, commonHeight, byzantineValidators)`:
 - `ev(B2, T, {v3, v4})` to `node1`
 - `ev(B1, T, {v2, v3})` to `node2`

Hermes reads them one at a time and, for each, creates two IBC headers of this form `H(valSet, trustedHeight, trustedVals)`. In our particular example they will be:
- `H1({v2, v3}, 100, {v1, v2, v3, v4}`
- `H2({v3, v4}, 100, {v1, v2, v3, v4}`

Hermes will end up sending two messages and ICS will do the following:
- `mis(H1, H2)`
   - determines the vals intersection between H2 (`{v3, v4}`) and the "trustedVals" (`{v1, v2, v3, v4}`)
   - the intersection `{v3, v4}` will be jailed
- `mis(H2, H1)`
   - determines the vals intersection between H2 (`{v2, v3}`) and the "trustedVals" (`{v1, v2, v3, v4}`)
   - the intersection `{v2, v3}` will be jailed

We end up with `{v2, v3, v4}` vals being jailed instead of `{v3}` which signed both `B1` and `B2`.

Reply:
> The LC agent will submit two evidences of form ev(conflictingBlock, commonHeight, byzantineValidators)

But from consensus, only one of the two evidences should be committed to a consumer chain block right? In this case, Hermes should only see the evidence that got persisted, which is assumed to report the correct conflicted header.

Reply:
In case of a fork a light client will submit evidence to each branch ([here](https://github.com/cometbft/cometbft/blob/17c32f980200e0dd36e348085e3300688c126c5d/light/detector.go#L243) and [here](https://github.com/cometbft/cometbft/blob/17c32f980200e0dd36e348085e3300688c126c5d/light/detector.go#L271)). Different hermes instances may/ will read the evidence from nodes on each therefore submitting two ICS misbehaviour messages.

Reply:
> Hermes will end up sending two messages

Is this because Hermes doesn't care which header is "correct", but rather that there are two conflicting headers at the same height and thus, the light client is no longer safe?